assets/components/dialog/dialog.jsx

```diff
@@ -0,0 +1,115 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React, { Component, type Node } from 'react';
+
+import { type Option } from 'helpers/types/option';
+import { classNameWithModifiers } from 'helpers/utilities';
+
+import './dialog.scss';
+
+
+// ----- Props ----- //
+
+export type PropTypes = {|
+  onStatusChange: (boolean) => void,
+  modal: boolean,
+  open: boolean,
+  'aria-label': Option<string>,
+  dismissOnBackgroundClick: boolean,
+  children: Node
+|};
+
+
+// ----- Component ----- //
+
+class Dialog extends Component<PropTypes> {
+
+  static defaultProps = {
+    onStatusChange: () => {},
+    modal: true,
+    open: false,
+    dismissOnBackgroundClick: false,
+  }
+
+  componentDidMount() {
+    if (this.props.open) {
+      this.open();
+    }
+  }
+
+  componentDidUpdate(prevProps: PropTypes) {
+    if (prevProps.open === true && this.props.open === false) {
+      this.close();
+    } else if (prevProps.open === false && this.props.open === true) {
+      this.open();
+    }
+  }
+
+  open() {
+    if (this.ref && this.ref.showModal) {
+      if (this.props.modal) {
+        this.ref.showModal();
+      } else {
+        this.ref.show();
+      }
+    }
+    requestAnimationFrame(() => {
+      if (this.ref) {
+        this.ref.focus();
+      }
+    });
+  }
+
+  close() {
+    if (this.ref && this.ref.close) {
+      this.ref.close();
+    }
```

Comment:
What happens to the focus after the modal is dismissed? Ideally it would return back to the button that opened it, or perhaps returned to the beginning of the page

Reply:
alas it depends on the implementer. the dialog does fire an event when it closes, so the component that fired it should be able to return the focus. In my testing it would seem that chrome makes a decent job at doing this on its own.

Something we could do could be to tweak the API so there's a nullable but required `returnFocusAfterClose()` prop to make it explicit that if you implement a dialog you have to take care of that? But I want to test default browser behaviour a bit more in case that just adds extra work

Reply:
Cool, sounds like it would work!

As an aside, it's worth testing the following combos of [browsers and screenreaders](https://webaim.org/projects/screenreadersurvey7/#browsercombos), as these are the most commonly used:

- IE11 + JAWS
- Firefox + NVDA
- Firefox + JAWS
- Safari + Voiceover